crates/sui-verifier-transactional-tests/tests/private_generics/receive_with_and_without_store.mvir

```diff
@@ -0,0 +1,38 @@
+//# publish
+module 0x0.m {
+    import 0x2.transfer;
+    import 0x2.object;
+
+    receive_good<T: key>(m: &mut object.UID, r: transfer.Receiving<T>): T {
+        let x: T;
+        label l0:
+        x = transfer.receive<T>(move(m), move(r));
+        return move(x);
+    }
+}
+
+//# publish
+module 0x0.m1 {
+    import 0x2.transfer;
+    import 0x2.object;
+
+    receive_good<T: key + store>(m: &mut object.UID, r: transfer.Receiving<T>): T {
+        let x: T;
+        label l0:
+        x = transfer.receive<T>(move(m), move(r));
+        return move(x);
+    }
+}
+
+//# publish
+module 0x0.m2 {
+    import 0x2.transfer;
+    import 0x2.object;
+
+    receive_bad<T>(m: &mut object.UID, r: transfer.Receiving<T>): T {
+        let x: T;
+        label l0:
+        x = transfer.receive<T>(move(m), move(r));
+        return move(x);
+    }
+}
```

Comment:
Small nit, I tend to find the transactional tests easier to reason about if you split  the valid/invalid cases into separate files